pkg/apis/componentconfig/v1alpha1/temp_v1a1_refactor.gogo

```diff
@@ -0,0 +1,548 @@
+  RemoteRuntimeEndpoint string `json:"remoteRuntimeEndpoint"`
+  // remoteImageEndpoint is the endpoint of remote image service
+  RemoteImageEndpoint string `json:"remoteImageEndpoint"`
+  // runtimeRequestTimeout is the timeout for all runtime requests except long running
+  // requests - pull, logs, exec and attach.
+  RuntimeRequestTimeout metav1.Duration `json:"runtimeRequestTimeout"`
+}
+
+/* BEGIN RUNTIME REMOVAL */
+
+// I believe the plan is to remove ALL of these from the Kubelet/Node configuration
+// e.g. they become flags-only for now and are immediately marked deprecated, for later removal.
+
+// Note: The WIP PR for making these flags-only is #44061
+
+type RuntimeOptions struct {
+  // networkPluginName is the name of the network plugin to be invoked for
+  // various events in kubelet/pod lifecycle
+  NetworkPluginName string `json:"networkPluginName"`
+  // networkPluginDir is the full path of the directory in which to search
+  // for network plugins (and, for backwards-compat, CNI config files)
+  NetworkPluginDir string `json:"networkPluginDir"`
+  // networkPluginMTU is the MTU to be passed to the network plugin,
+  // and overrides the default MTU for cases where it cannot be automatically
+  // computed (such as IPSEC).
+  NetworkPluginMTU int32 `json:"networkPluginMTU"`
+  // CNIConfDir is the full path of the directory in which to seardch for
+  // CNI config files
+  CNIConfDir string `json:"cniConfDir"`
+  // CNIBinDir is the full path of the directory in which to search for
+  // CNI plugin binaries
+  CNIBinDir string `json:"cniBinDir"`
+  // podInfraContainerImage is the image whose network/ipc namespaces
+  // containers in each pod will use.
+  PodInfraContainerImage string `json:"podInfraContainerImage"`
+}
+
+type DockerOptions struct {
+  // dockerEndpoint is the path to the docker endpoint to communicate with.
+  DockerEndpoint string `json:"dockerEndpoint"`
+  // enable gathering custom metrics.
+  EnableCustomMetrics bool `json:"enableCustomMetrics"` // for Docker
+}
+
+type RktOptions struct {
+  // rktPath is the  path of rkt binary. Leave empty to use the first rkt in
+  // $PATH.
+  RktPath string `json:"rktPath"`
+  // rktApiEndpoint is the endpoint of the rkt API service to communicate with.
+  RktAPIEndpoint string `json:"rktAPIEndpoint"`
+}
+
+/* END RUNTIME REMOVAL */
+
+type NodeRegistrationConfig struct {
+    // registerNode enables automatic registration with the apiserver.
+  RegisterNode *bool `json:"registerNode"`
+  // registerWithTaints are an array of taints to add to a node object when
+  // the kubelet registers itself. This only takes effect when registerNode
+  // is true and upon the initial registration of the node.
+  RegisterWithTaints []v1.Taint `json:"registerWithTaints"`
+  // nodeLabels to add when registering the node in the cluster.
+  NodeLabels map[string]string `json:"nodeLabels"`
+
+}
+
+type NodeStatusConfig struct {
```

Comment:
Do we really need separate structs for `NodeRegistrationConfig` and `NodeStatusConfig`?